📖 Add explicit mention of working groups as owners (#30137)
ampproject/amphtml: contributing/contributing-code.md

We now have a bot that will automatically assign Owners to a PR once it is created, and it is likely at least one of these Owners will also be a Reviewer.

> Depending on the files/folders involved, Owners include one or more [AMP Working Groups](https://amp.dev/community/governance/#working-groups), such as the groups for [Security & Privacy](https://amp.dev/community/working-groups/security-privacy/) and [User Interface & Accessibility](https://amp.dev/community/working-groups/ui-and-a11y/).

Reviewer: To make this a bit more aligned with the intention described in the PR description, I might tweak this a bit
```suggestion
Depending on the files/folders involved, Owners include one or more [AMP Working Groups](https://amp.dev/community/governance/#working-groups), which may include the [Security & Privacy](https://amp.dev/community/working-groups/security-privacy/) and [User Interface & Accessibility](https://amp.dev/community/working-groups/ui-and-a11y/) WGs.
```

Author: "Owners *include* ... which may *include*" sounds, subjectively, a bit awkward. I think the meaning/intention is already clear in the current wording (the "such as" suggesting it's an example of the groups that may be included). However, feel free to tweak the wording before merging.

Reviewer: Maybe something like"Most files and folders are owned by one or more AMP Working Groups, which may include the..."

The way it's written currently does not make it clear why those working groups are being called out. We could even go a step more explicit saying something more along the lines of "Many files and folders are owned by the UI & A11y Working Group, which can review for accessibility.". I don't see much value in mentioning the Security & Privacy WG here at all, as it's not too common for them to be listed as Owners.

Reviewer 2: The confusion may have to do with the term "Owners" -- There are literal folder-based `OWNERS` where reviews are required by a certain working group, as well as code changes of a certain _type_ that would be best made if done in consultation with a working group owning a certain knowledge domain. 

This is why it made sense to me that UI & A11y  and Security & Privacy are called out specifically for their domain areas which may be relevant regardless of code location. In that regard it makes sense to call it out explicitly in documentation instead of via other automated ownership processes because whether a review might be needed for these considerations is not clearly deterministic.

Reviewer: Given that, I feel like the Owners section isn't the best place for such a callout to land. This part of the docs is focused around explaining that owners will be assigned in general by the bot. I think elsewhere in the doc might be more appropriate to mention that some changes should get a11y or security/privacy reviews, but I feel like that doesn't tie into the Owners system directly.

Reviewer 2: The current "Code review and Approval" seems like the most appropriate location for this text. The Owners system is described in a different section called "Roles". We can update the text as is to be more clearly distinct:
```
Please note that some Owners may be relevant regardless of code location and therefore may 
include one or more AMP Working Groups such as the groups for Security & Privacy and 
User Interface & Accessibility. 
```

Reviewer: +1 to the "Code Review & Approval" section suggestion. That updated text LGTM, thanks @caroqliu

Author: Moved the text to the note as suggested by @caroqliu

Reviewer 2: Thank you! @rcebulko can you help with CLA again? And then we will be good to merge.

Reviewer 2: Actually it looks like prettier is complaining too: @TetraLogicalHelpdesk https://travis-ci.org/github/ampproject/amphtml/jobs/729223189

> Depending on the files/folders involved, Owners include one or more [AMP Working Groups](https://amp.dev/community/working-groups/access-subscriptions/), such as the groups for  [Security & Privacy](https://amp.dev/community/working-groups/security-privacy/) and [User Interface & Accessibility](https://amp.dev/community/working-groups/ui-and-a11y/).

Reviewer 2: This is a better link for `AMP working groups`: https://amp.dev/community/governance/#working-groups

Author: Ah, perfect. Yes that's a much better link (I was going by the link in the navigation that you get to for "Community > Working Groups" - it might be worth, separately, setting up a high-level page that's generic for all WGs there, rather than linking straight to the first WG)

  - Owners of an area may approve other Owners at or below their area of expertise following the normal PR process. To request becoming an Owner create a PR adding yourself to the appropriate OWNERS file and assign/cc a current Owner for that directory.
- The list of Owners for a directory can be found in the [OWNERS](https://github.com/ampproject/amphtml/search?o=asc&q=filename%3A"OWNERS"&s=indexed) file in the directory or a parent directory.

> Please note that some Owners may be relevant regardless of code location and therefore may include one or more [AMP Working Groups](https://amp.dev/community/governance/#working-groups), such as the groups for [Security & Privacy](https://amp.dev/community/working-groups/security-privacy/) and [User Interface & Accessibility](https://amp.dev/community/working-groups/ui-and-a11y/). 

Reviewer 2: Looks like prettier is complaining about the space at the end of the line -- apologies for the delay. 

Once the checks run again I'd be happy to merge. :) 

@rcebulko what is the deal with CLA? Can we just update the label or is there more to it? 
```suggestion
Please note that some Owners may be relevant regardless of code location and therefore may include one or more [AMP Working Groups](https://amp.dev/community/governance/#working-groups), such as the groups for [Security & Privacy](https://amp.dev/community/working-groups/security-privacy/) and [User Interface & Accessibility](https://amp.dev/community/working-groups/ui-and-a11y/).
```

Reviewer: This is the new OpenJS CLA, the old Google CLA bot is deprecated now that AMP is part of JSF

Reviewer: See https://github.com/ampproject/meta/issues/60#issuecomment-705851924

Reviewer 2: I believe the new OpenJS CLA is still non-blocking. It looksl ike the Google CLA is still what is needed to submit this PR. :)

Reviewer 2: @patrickhlauke @TetraLogicalHelpdesk The PR also needs the suggested change above (remove white space) :) to pass our CI, thanks!
